pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -441,17 +434,93 @@ func setNodeScoresToZero(nodes []v1.Node) (priorityList schedulerapi.HostPriorit
 	return
 }
 
+func getRequestedVolumeSet(ns string, requestedVolumes []v1.Volume) (volumeSet map[string]struct{}, volumeCount int) {
+	// create a set of all volumes needed
+	volumeSet = map[string]struct{}{}
+	for _, volume := range requestedVolumes {
+		// not the same as len(pvNamesForRequestedVolumes). PV-s for a new pod might not be cached yet.
+		volumeCount++
```

Comment:
This is not in the right place. Currently this is the same as len(requestedVolumes). Only count the volumes that have `PersistentVolumeClaim` defined (i.e. line 444), otherwise you include the default vol for the secret which we don't care about.

Reply:
Good catch. Made a mistake there.